packages/utils/src/internals/sitemap.ts

```diff
@@ -0,0 +1,103 @@
+import { createGunzip } from 'node:zlib';
+
+import log from '@apify/log';
+import type { SAXStream } from 'sax';
+import sax from 'sax';
+
+class ParsingState {
+    sitemapUrls: string[] = [];
+    urls: string[] = [];
+    visitedSitemapUrls: string[] = [];
+    context: 'sitemapindex' | 'urlset' | undefined = undefined;
```

Comment:
should be the same as

```suggestion
    context?: 'sitemapindex' | 'urlset';
```